### PR TITLE
[Project Addition] : Fake and Real Face Detection using DL

### PR DESCRIPTION
**Pull request by aaradhyasinghgaur**
# Pull Request for DL-Simplified 💡

## Issue Title : Fake and Real Face Detection using DL 

- **Info about the related issue (Aim of the project)** :  To identify fake and real faces which present a harder challenge to classify them correctly even for the human eye.
- **Name:**  Aaradhya Singh 
- **GitHub ID:** https://github.com/kyra-09 
- **Email ID:**  aaradhyasinghgaur@gmail.com 
- **Idenitfy yourself: (Mention in which program you are contributing in. Eg. For a JWOC 2022 participant it's, `JWOC Participant`)**  GSSOC-2024 Contributor 

Closes: #725

### Describe the add-ons or changes you've made 📃

  1.  Data Preparation: Collect and preprocess a balanced dataset of real and fake face images, including normalization, resizing, and augmentation.
  2.  Base Model Selection: EfficientNetB0,VGG16 , Xception , InceptionV3 like 5 different models excluding its top layers, to leverage its learned features.
   3.  Model Construction: Add custom layers on top of the base model for binary classification, compiling with appropriate loss and metrics.
 4.   Initial Training: Train the model with the base layers frozen to only update the new layers.
  5.  Fine-Tuning: Unfreeze some or all of the base model layers and continue training with a lower learning rate to fine-tune the entire network.
    6.) EDA analysis.
    7.) Comaprioson using performance matrices such as accuracy scores , confusion matrix etc.

## Type of change ☑️

What sort of change have you made:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested? ⚙️

Describe how it has been tested
Describe how have you verified the changes made

## Checklist: ☑️

- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [x] I have added things that prove my fix is effective or that my feature works.
- [ ] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

**Comment by abhisheks008** [comment]
Hi @kyra-09 please update the project folder name as I have updated the issue name. Please do the needful.

**Comment by kyra-09** [comment]
okay.. i'll do it right now.

**Comment by kyra-09** [comment]
@abhisheks008 Thank you for your time and work . You have been most helpful and consistent with assigning and reviewing our PRs .

**Review by abhisheks008 — APPROVED** [review]
Hi @kyra-09 approved this PR.

Do not push your PRs in a continuos way, it may detected as a spam by the leaderboard algorithm. Put at least a gap of 12 hours between two PRs. Please keep this in mind.
